### PR TITLE
[GLUTEN-4122][VL] Add num_threads option to limit build threads for velox

### PR DESCRIPTION
**Pull request by wForget**
## What changes were proposed in this pull request?

Add NUM_THREADS argument to limit compilation parallelism

Fixes: #4122

### CODE REVIEW

File: dev/builddeps-veloxbe.sh [hunk 1]
```diff
@@ -29,6 +29,7 @@ VELOX_REPO=""
 VELOX_BRANCH=""
 VELOX_HOME=""
 VELOX_PARAMETER=""
+NUM_THREADS=""
```
Comment (dcoliversun) [inline dev/builddeps-veloxbe.sh:32]: maybe we need default value here `$(nproc --ignore=2)`

Comment (wForget) [reply dev/builddeps-veloxbe.sh:32]: > maybe we need default value here `$(nproc --ignore=2)`

sgtm

**Review by dcoliversun — APPROVED** [review]
lgtm

**Comment by PHILO-HE** [comment]
Thanks for your PR! Please do a code rebase and resolve the code conflict.

**Comment by wForget** [comment]
> Thanks for your PR! Please do a code rebase and resolve the code conflict.

Sorry for replying so late due to being busy with some things. I resolved the conflict.

File: dev/builddeps-veloxbe.sh [hunk 2]
```diff
@@ -32,6 +32,7 @@ VELOX_BRANCH=""
 VELOX_HOME=""
 VELOX_PARAMETER=""
 COMPILE_ARROW_JAVA=OFF
+NUM_THREADS=$(nproc --ignore=2)
```
Comment (PHILO-HE) [inline dev/builddeps-veloxbe.sh:35]: Maybe, someone prefers directly setting NUM_THREADS variable in the env. Can we set `NUM_THREADS` here only if it is not set? Thanks!

Comment (wForget) [reply dev/builddeps-veloxbe.sh:35]: Fixed, thanks.

Comment (wForget) [reply dev/builddeps-veloxbe.sh]: Fixed, thanks.

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**